app/survey_config/business_config.py

```diff
@@ -48,8 +60,20 @@ def __post_init__(self):
     def get_service_links(
         self, sign_out_url: str, *, is_authenticated: bool
     ) -> Optional[list[dict]]:
+        if self.schema and is_authenticated:
+            ru_ref = get_metadata(current_user).get("ru_ref")  # type: ignore
```

Comment:
This is still using app context. We should pass in ru_ref.

Reply:
So I moved `ru_ref` all the way to template helpers' as an arg of `get_survey_links`, I hope you don't mind schema here as it is not present in template helpers.